src/Microsoft.AspNetCore.SystemWebAdapters/Internal/SessionStateFeature.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Web.SessionState;
+using Microsoft.AspNetCore.SystemWebAdapters.SessionState;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Internal
+{
+    internal class SessionStateFeature: ISessionStateFeature
```

Comment:
I would pass in the httpcontext and default to the endpoint metadata if Feature.State has not been set

Reply:
The SessionAttribute and SessionBehavior  are in CoreServices, while the feature is in SystemWebAdapters. 
I could move them to SystemWebAdapters or try creating interface and use interface in the SystemWebAdapters.
Any thought?

Reply:
yup - those layerings are why we end up use features for things :) here's a branch that took what you did and moved it to different layers: https://github.com/dotnet/systemweb-adapters/tree/tasou/sessionbehavior. note as part of this, I'm deprecating parts of the previous attribute to use this behavior mechanism throughout the stack more. Not sure if it works fully, but feel free to take it or ideas from it.

Reply:
The change in https://github.com/dotnet/systemweb-adapters/tree/tasou/sessionbehavior looks good to me.
I have two comments for it:
1. Creating SessionAttribute looks weird in SessionLoadMiddleware.ManageStateAsync. If you only want to use the feature, why creating sessionAttribute?
2. Could you add a test for SetSessionStateBehavior?

Will you create a PR for the branch? If you are going to, I could abandon this PR.

Reply:
I'll let you make the final PR - that was just an attempt at how to handle the layering.

> Creating SessionAttribute looks weird ...

Yes, it does. However, for whatever reason, I made ISessionManager take a SessionAttribute and it's a public API :( so we're kind of stuck with that pattern for now.

Reply:
@twsouthwick  I add a new test in test/Microsoft.AspNetCore.SystemWebAdapters.CoreServices.Tests/SessionState/SetSessionStateBehaviorTests.cs. But I will get error when setting session is readonly. Do you have any thoughts how to solve it?

error:
```
Message: 
System.Net.Http.HttpRequestException : Error while copying content to a stream.
---- System.IO.IOException : 
-------- System.InvalidOperationException : Session is readonly

  Stack Trace: 
HttpContent.LoadIntoBufferAsyncCore(Task serializeToStreamTask, MemoryStream tempBuffer)
HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
SetSessionStateBehaviorTests.GetResponseAsync(HttpClient client, String uri) line 91
SetSessionStateBehaviorTests.TestSetSessionStateBehavior() line 65
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
Pipe.GetReadResult(ReadResult& result)
Pipe.GetReadAsyncResult()
DefaultPipeReader.GetResult(Int16 token)
ResponseBodyReaderStream.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
Stream.<CopyToAsync>g__Core|29_0(Stream source, Stream destination, Int32 bufferSize, CancellationToken cancellationToken)
StreamToStreamCopy.<CopyAsync>g__DisposeSourceAsync|1_0(Task copyTask, Stream source)
HttpContent.LoadIntoBufferAsyncCore(Task serializeToStreamTask, MemoryStream tempBuffer)
----- Inner Stack Trace -----
AspNetCoreSessionState.CheckReadOnly() line 42
AspNetCoreSessionState.CommitAsync(CancellationToken token) line 122
SessionLoadMiddleware.ManageStateAsync(HttpContext context, ISessionStateFeature feature) line 58
<<UseHttpApplicationEvent>b__0>d.MoveNext() line 95
--- End of stack trace from previous location ---
<<UseHttpApplicationEvent>b__0>d.MoveNext() line 95
--- End of stack trace from previous location ---
<<UseHttpApplicationEvent>b__0>d.MoveNext() line 95
--- End of stack trace from previous location ---
<<UseHttpApplicationEvent>b__0>d.MoveNext() line 95
--- End of stack trace from previous location ---
SessionStateMiddleware.InvokeAsync(HttpContext context) line 20
PreBufferRequestStreamMiddleware.InvokeAsync(HttpContext context) line 31
HttpApplicationMiddleware.InvokeAsync(HttpContext context) line 41
HttpApplicationMiddleware.InvokeAsync(HttpContext context) line 46
SessionMiddleware.Invoke(HttpContext context)
SessionMiddleware.Invoke(HttpContext context)
RegisterAdapterFeaturesMiddleware.InvokeAsync(HttpContext context) line 28
RegisterAdapterFeaturesMiddleware.InvokeAsync(HttpContext context) line 33
<<SendAsync>g__RunRequestAsync|0>d.MoveNext()
```

Reply:
looks like it was because you were committing the session when it was marked as readonly. I added the check in for you